### PR TITLE
Add Android string resource support

### PR DESCRIPTION
**Pull request by eemeli**
Handles all parts of the resource format, except for ICU MessageFormat strings.

### CODE REVIEW

File: tests/data/strings.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  <!-- standalone -->
+
+  <string name="welcome">Welcome to <b>&foo;</b>!</string>
+  <string name="placeholders">Hello, %1$s! You have %2$d new messages.</string>
+  <string name="escape_html">Hello, %1$s! You have &lt;b>%2$d new messages&lt;/b>.</string>
+
+  <string name="ws_trimmed"> &#32; &#8200; &#8195;</string>
+  <string name="ws_quoted">" &#32; &#8200; &#8195;&quot;</string>
+  <string name="ws_escaped"> \u0032 \u8200 \u8195</string>
+
```
Comment (mathjazz) [inline tests/data/strings.xml:25]: I would also expect apostrophes to be tested and handled properly:
https://github.com/mozilla/pontoon/pull/3020/files#diff-e8bac283d0aa4db32a8a4a7f254bc66cadeeb611975a0d1beb8d5ecb4979dc80R324

File: tests/data/strings.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <item>Mars</item>
+  </string-array>
+
+  <plurals name="numberOfSongsAvailable">
+    <!--
+         As a developer, you should always supply "one" and "other"
+         strings. Your translators will know which strings are actually
+         needed for their language. Always include %d in "one" because
+         translators will need to use %d for languages where "one"
+         doesn't mean 1 (as explained above).
```
Comment (mathjazz) [inline tests/data/strings.xml:44]: Where "above"?

**Review by mathjazz — COMMENTED** [review]
This is impressive. Android plurals!

File: tests/data/strings.xml (hunk 2 above)
Comment (eemeli) [reply tests/data/strings.xml:44]: Ah, a remnant from the [Android docs](https://developer.android.com/guide/topics/resources/string-resource.html#Plurals) from where I lifted this.

**Review by mathjazz — APPROVED** [review]

**Comment by eemeli** [comment]
Alas, needed to refactor this still a bunch to properly support `<xliff:g>`, which can be used to mark parts of the message as nontranslatable.

Also, it turns out that at least some of the Android escaping is valid within nested elements. But not all of it.

Please take another look?

**Review by mathjazz — APPROVED** [review]
The code looks good, although I found it a bit tricky to review, so I'm mostly relied on the tests. I'm confident we'll spot any issues as part of integrating this code into Pontoon and testing with the real files.

I'd only add that link to the <xliff:g> reference.

File: moz/l10n/android/parse.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+#   - Strings with ICU MessageFormat contents.
+#     These also use "quotes" for special escaping behaviour.
+#     ICU MessageFormat strings are not currently detected by this library.
+#
+# The source contents of each of the above needs to be parsed differently,
+# and message strings can be found in <string>, <string-array>, and <plurals>
+# elements, each of which also needs different parsing.
+#
+# For more information, see:
+# https://developer.android.com/guide/topics/resources/string-resource
```
[the remark this answers is not in the document]
Comment (eemeli) [reply moz/l10n/android/parse.py:59]:
```suggestion
# https://developer.android.com/guide/topics/resources/string-resource
# https://developer.android.com/guide/topics/resources/localization#mark-message-parts
```

**Pull request merged**